packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -242,8 +263,11 @@ export class SummaryGenerator {
 			{ start: true, end: true, cancel: "generic" },
 		);
 
+		// Use record type to prevent unexpected value types
+		let summaryData: SubmitSummaryResult | undefined;
 		const fail = (
 			errorCode: keyof typeof summarizeErrors,
+			stage: SummaryStage = summaryData?.stage ?? "unknown",
```

Comment:
This is kind of creepy to have as a param but also have the "global" fallback logic, who is overriding this default logic?

Reply:
Looking below, no one passes anything besides `undefined` or `summaryData.stage` which are equivalent, right?  Can we remove this param?

Reply:
Good catch. Removed it.